chain/chain/src/store/latest_witnesses.rs

```diff
@@ -118,23 +133,36 @@ impl ChainStore {
             .get_ser::<LatestWitnessesInfo>(DBCol::Misc, LATEST_WITNESSES_INFO)?
             .unwrap_or_default();
 
+        let new_witness_index = info.next_witness_index;
+
         // Adjust the info to include the new witness.
         info.count += 1;
         info.total_size += serialized_witness.len() as u64;
+        info.next_witness_index += 1;
 
-        // Go over witnesses with increasing (height, shard_id) and remove them until the limits are satisfied.
-        // Height and shard id are stored in big-endian representation, so sorting the binary representation is
-        // the same as sorting the integers.
         let mut store_update = self.store().store_update();
-        for item in self.store().iter(DBCol::LatestChunkStateWitnesses) {
-            if info.is_within_limits() {
-                break;
-            }
 
-            let (key_bytes, witness_bytes) = item?;
-            store_update.delete(DBCol::LatestChunkStateWitnesses, &key_bytes);
+        // Go over witnesses with increasing indexes and remove them until the limits are satisfied.
+        while !info.is_within_limits() && info.lowest_index < info.next_witness_index {
```

Comment:
Oh wait I see that this is debug-only and not recommended for production use. Okay, maybe this is not a big deal then.

Reply:
Yeah it could potentially take a long time if it needs to remove a lot of saved witnesses :/
In the average case it only removes one or two, but in the worst case scenario it could need to remove thousands of witnesses.

It's a debug-only optional feature, and I'm not sure if anyone is really using it, so I didn't spend too much time optimizing things. We can fix it in a follow up PR if it becomes a problem.